libraries/ms-common/inc/pwm.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+// PWM Library
+//
+// Usage Notes:
+// - Module must be initialized before use.
+// - Use pwm_set_dc unless you have reason not to.
+// - pwm_set_pulse affords resolution < 1% of the duty cycle.
+// - For stm32f0xx GPIO pins must use the correct GPIO_ALTFN to utilize the PWM.
```

Comment:
[I don't think so looking at the alternate function GPIO pin tables in here.](http://www.st.com/content/ccc/resource/technical/document/datasheet/cd/46/43/83/22/d3/40/c8/DM00090510.pdf/files/DM00090510.pdf/jcr:content/translations/en.DM00090510.pdf).

In summary, you need to set the alternate function on the GPIO pin to the correct alternate function channel to connect them to the PWM pulse from the timer. Since each TIM a pin is tied to has its own alternative function so far as I can see in the data sheet we don't need to worry about this. 

I've tested this on the discovery and even with TIM3 enabled on all channels doing PWM the GPIOs for the LEDs only use PWM if I explicitly set them to do so via their alternative function.